python/mxnet/gluon/block.py

```diff
@@ -954,6 +955,18 @@ def _build_cache(self, *args):
             unused = ', '.join(list(param_names - set(used_param_names)))
             warnings.warn("Parameter %s is not used by any computation. "
                           "Is this intended?"%unused, stacklevel=4)
+        if self._backend:
+            arg_array = []
+            ctx = None
```

Comment:
I think we only need to assign `ctx` to `args[0].context` since in gluon there must be at least one input to the model and all inputs/params have to be on the same context. Then we dont need to do it in the loop.

Reply:
Fixed